Enable X86 / Neon flags only for that architecture.

Set ARCH_DEF here.

--- variables.am.orig	2020-03-16 08:25:44.000000000 +0000
+++ variables.am
@@ -1,5 +1,6 @@
 # Another arch can be forced here
 ARCH_NAME = $(ARCH_DIR)
+ARCH_DEF = $(ARCH_DIR)
 
 LANGFLAGS = -std=c99 -fno-strict-aliasing
 LANGXXFLAGS = -fno-strict-aliasing
@@ -13,10 +14,14 @@ INCFLAGS += -I$(top_srcdir)/src/include/
 
 OFLAGS ?= -O2
 
-if X86
-ARCHFLAGS = -maes -mavx2 -mbmi2 #TODO
+if ARCH_X86
+ARCHFLAGS = -maes #-mavx2 -mbmi2 #TODO
 else
-ARCHFLAGS = # -mavx2 -mbmi2 #TODO
+if ARCH_NEON
+ARCHFLAGS = -mfpu=neon
+else
+ARCHFLAGS = #
+endif
 endif
 
 ARCHFLAGS += $(XARCHFLAGS)
